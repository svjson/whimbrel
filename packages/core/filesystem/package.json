{
  "name": "@whimbrel/filesystem",
  "version": "0.0.2",
  "description": "File System abstractions for Whimbrel",
  "main": "./dist/index.cjs",
  "module": "./dist/index.js",
  "types": "./dist/index.d.ts",
  "exports": {
    ".": {
      "require": "./dist/index.cjs",
      "import": "./dist/index.js"
    }
  },
  "files": [
    "dist"
  ],
  "scripts": {
    "build": "tsup",
    "test": "vitest run",
    "typecheck": "tsc --noEmit",
    "clean": "rimraf ./dist",
    "publish": "npm publish --access public"
  },
  "peerDependencies": {
    "@whimbrel/core-api": "^0.0.2"
  },
  "dependencies": {
    "@whimbrel/core-api": "*",
    "jsonc-parser": "^3.3.1"
  },
  "devDependencies": {
    "@whimbrel-test/tree-fixtures": "*",
    "@types/node": "^24.3.0",
    "vite": "^7.1.2",
    "vite-tsconfig-paths": "^5.1.4",
    "vitest": "^3.2.4"
  }
}
