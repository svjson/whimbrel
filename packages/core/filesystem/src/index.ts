export { ContextFileSystem } from './context'
export { DiskFileSystem, PhysicalFileSystem } from './physical'
export { MemoryFileSystem, makeMemoryFileSystem } from './memfs'
export { FileSystemReporter, FileSystemMutationReporter } from './reporter'
export { ReadThroughFileSystem, makeReadThroughMemoryFileSystem } from './read-through'
export { makeNodeFsAdapter } from './node-fs-adapter'
export { makeNodeFsPromisesAdapter } from './node-fs-promises-adapter'
export type { NodeFileSystem, NodePromisesFileSystem } from './types'
