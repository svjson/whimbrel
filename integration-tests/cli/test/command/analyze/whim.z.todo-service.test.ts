import { describe, it, expect, beforeAll, test } from 'vitest'
import makeTreeFixture from '@whimbrel-test/tree-fixtures'
import { DiskFileSystem } from '@whimbrel/filesystem'
import { whimCli } from 'test/whim-fixture'

const { prepareGitRepository } = makeTreeFixture(DiskFileSystem)

describe('whim z <todo-service.ts>', () => {
  let root: string

  beforeAll(async () => {
    root = await prepareGitRepository(DiskFileSystem, 'todo-service')
  })

  test('whim z', async () => {
    const whim = whimCli(root)

    await whim.execute('z')

    expect(whim.stdout).toMatchSnapshot()
    expect(whim.stdout).toContain('Define Source')
    expect(whim.stdout).toContain('Analyze Actor')
    expect(whim.stdout).toContain('Discover Actor Facets')
    expect(whim.stdout).toContain('Reify Actor')
  })

  test('whim z --no-color', async () => {
    const whim = whimCli(root)

    await whim.execute(['z', '--no-color'])

    expect(whim.stdout).toMatchSnapshot()
    expect(whim.stdout).toContain('Define Source')
    expect(whim.stdout).toContain('Analyze Actor')
    expect(whim.stdout).toContain('Discover Actor Facets')
    expect(whim.stdout).toContain('Reify Actor')
  })
})
