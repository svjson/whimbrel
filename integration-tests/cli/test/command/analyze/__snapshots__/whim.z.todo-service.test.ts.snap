// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`whim z <todo-service.ts> > whim z --no-color 1`] = `
"
Analyze path: /tmp/whim-repo-XXXX
-----------------------------------
* Define Source
  âŠ• source: whim-repo-XXXX
  * Analyze Actor
    * Discover Actor Facets
      â‡’ vcs: git
      â‡’ engine: node
      â‡’ package-file: package.json
      â‡’ facets: project
  * Reify Actor
    â‡’ name: todo-service

"
`;

exports[`whim z <todo-service.ts> > whim z 1`] = `
"
Analyze path: [32m/tmp/whim-repo-XXXX[39m
-----------------------------------
* Define Source
  [32mâŠ•[39m [35msource[39m: [36mwhim-repo-XXXX[39m
  * Analyze Actor
    * Discover Actor Facets
      [36mâ‡’[39m [34mvcs[39m: [36mgit[39m
      [36mâ‡’[39m [34mengine[39m: [36mnode[39m
      [36mâ‡’[39m [34mpackage-file[39m: [36mpackage.json[39m
      [36mâ‡’[39m [34mfacets[39m: [36mproject[39m
  * Reify Actor
    [36mâ‡’[39m [35mname[39m: [36mtodo-service[39m

"
`;
